shoop/campaigns/migrations/0006_campaign_effects.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import shoop.core.fields
+
+cached_values = []
+
+# FORWARDS
+def cache_values(apps, schema_editor):
+    print("\n--- Caching values ---")
+    CatalogCampaign = apps.get_model("campaigns", "CatalogCampaign")
+    BasketCampaign = apps.get_model("campaigns", "BasketCampaign")
+    db_alias = schema_editor.connection.alias
+
+    def cache_value(c, cache_type):
+        if not c.discount_amount_value and not c.discount_percentage:
+            return
+        cached_values.append({
+            "id": c.id,
+            "type": cache_type,
+            "amount": c.discount_amount_value,
+            "percentage": c.discount_percentage
+        })
```

Comment:
"Caching" these to memory might be bad idea, because (a) there could be so many campaigns that there is not enough memory and (b) if the migration is aborted for some reason, the values will be lost. Even transaction won't help since migration invokes DDL statements which mess up at least MySQL's transactions.